webapp/src/components/PayMortgagePage/PayMortgageForm/PayMortgageForm.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { Button, Form, Input } from 'semantic-ui-react'
+import { preventDefault } from 'lib/utils'
+import { t } from 'modules/translation/utils'
+import { mortgageType } from 'components/types'
+
+import './PayMortgageForm.css'
+
+export default class PayMortgageForm extends React.PureComponent {
+  static propTypes = {
+    balance: PropTypes.number,
+    mortgage: mortgageType,
+    onSubmit: PropTypes.func.isRequired,
+    onCancel: PropTypes.func.isRequired,
+    isDisabled: PropTypes.bool.isRequired
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      amount: ''
+    }
+  }
+
+  handleAmountChange = e => {
+    const { mortgage, balance } = this.props
+    const amount = e.currentTarget.value
+    if (amount === '') {
+      this.setState({ amount })
+    } else {
+      this.setState({
+        amount: Math.max(
+          Math.min(mortgage.outstanding_amount, balance, amount),
+          0
+        )
+      })
+    }
+  }
+
+  handleSubmit = () => {
+    const {
+      mortgage: { loan_id, asset_id },
+      onSubmit
+    } = this.props
```

Comment:
_You don't need to change this_: this is perfectly valid, but I lean towards using something like:

```javascript
const { mortgage, onSubmit } = this.props
const { loan_id, asset_id } = morgage
```

it accomplishes the same in fewer lines and it's consistent with the rest of the codebase